File: app/src/main/java/com/softdev/smartraysam/carfeeds/Core/GetDataContract.java
package com.softdev.smartraysam.carfeeds.Core;

import android.content.Context;

import com.softdev.smartraysam.carfeeds.model.carModel;

import java.util.List;



/**
 * Created by Smart Raysam on 15-09-2018.
 */

public interface GetDataContract {
    interface View{
        void onGetDataSuccess(String message, List<carModel> list);
        void onGetDataFailure(String message);
        void onStart(String message);
    }
    interface Presenter{
        void getDataFromURL(Context context, String url, String msg);
    }
    interface Interactor{
        void initRetrofitCall(Context context, String url);

    }
    interface onGetDataListener{
        void onSuccess(String message, List<carModel> list);
        void onFailure(String message);
    }
}

File: app/src/main/java/com/softdev/smartraysam/carfeeds/contract/GetDataContract.kt
package com.softdev.smartraysam.carfeeds.contract

import android.content.Context

import com.softdev.smartraysam.carfeeds.model.carModel


/**
 * Created by Smart Raysam on 15-09-2018.
 */

interface GetDataContract {
    interface View {
        fun onGetDataSuccess(message: String, list: List<carModel>)
        fun onGetDataFailure(message: String)
        fun onStart(message: String)
    }

    interface Presenter {
        fun getDataFromURL(context: Context, url: String, msg: String)
    }

    interface Interactor {
        fun initRetrofitCall(context: Context, url: String)

    }

    interface onGetDataListener {
        fun onSuccess(message: String, list: List<carModel>)
        fun onFailure(message: String)
    }
}
